core/serialisation/src/main/java/uk/gov/gchq/gaffer/serialisation/util/MultiSerialiserStorage.java

```diff
@@ -0,0 +1,233 @@
+ */
+public class MultiSerialiserStorage {
+    private Map<Byte, Class<? extends ToBytesSerialiser>> keyToSerialiser = new HashMap<>();
+    private Map<Byte, Class> keyToValueMap = new HashMap<>();
+    private Map<Class, Byte> valueToKeyMap = new HashMap<>();
+    private boolean consistent = true;
+    private boolean preservesObjectOrdering = true;
+    private byte savedSerialiserEncoding = ((byte) 0);
+
+    public void put(Class<? extends ToBytesSerialiser> serialiserClass, Class supportedClass) throws GafferCheckedException {
+        put(getSerialiserEncoding(), serialiserClass, supportedClass);
+    }
+
+    public void put(byte serialiserEncoding, Class<? extends ToBytesSerialiser> serialiserClass, Class supportedClass) throws GafferCheckedException {
+        ToBytesSerialiser toBytesSerialiser = getSerialiser(serialiserClass, supportedClass);
+
+        consistent = continuesToBeConsistant(toBytesSerialiser);
+        preservesObjectOrdering = continuesToPreserveOrdering(toBytesSerialiser);
+
+        keyToSerialiser.put(serialiserEncoding, serialiserClass);
+        keyToValueMap.put(serialiserEncoding, supportedClass);
+        valueToKeyMap.put(supportedClass, serialiserEncoding);
+    }
+
+    private ToBytesSerialiser getSerialiser(final Class<? extends ToBytesSerialiser> serialiserClass, final Class supportedClass) throws GafferCheckedException {
+        ToBytesSerialiser toBytesSerialiser;
+        try {
+            toBytesSerialiser = serialiserClass.newInstance();
+        } catch (Exception e) {
+            throw new GafferCheckedException(String.format("Unable to Instantiate serialiser to validate that %s can handle %s", serialiserClass.getSimpleName(), supportedClass.getSimpleName()), e);
+        }
+        if (!toBytesSerialiser.canHandle(supportedClass)) {
+            throw new GafferCheckedException(String.format("%s does not handle %s", toBytesSerialiser.getClass(), supportedClass));
+        }
+        return toBytesSerialiser;
+    }
+
+    private byte getSerialiserEncoding() {
+        Set<Byte> integers = keyToSerialiser.keySet();
+        while (integers.contains(savedSerialiserEncoding)) {
+            savedSerialiserEncoding++;
+        }
+        return savedSerialiserEncoding;
+    }
+
+    public ToBytesSerialiser getSerialiserFromKey(final byte key) throws GafferCheckedException {
+        Class<? extends ToBytesSerialiser> serialiserClass = keyToSerialiser.get(key);
+        try {
+            return (null == serialiserClass) ? null : serialiserClass.newInstance();
+        } catch (InstantiationException | IllegalAccessException e) {
+            throw new GafferCheckedException(String.format("Unable to Instantiate serialiser, this means that serialiser was not checked when added to the storage: %s", serialiserClass));
+        }
+    }
+
+    public ToBytesSerialiser getSerialiserFromValue(final Object object) throws GafferCheckedException {
```

Comment:
This is going to be quire inefficient. We don't want to construct new serialiser instances every time we need to serialise/deserialise something. We need to change the keyToSerialiser map to be a map of key to serialiser instances.